File: js/arrays/map/changeOriginal/ruby/closure.rb
# closure.rb

letters = ["a","b","c","d","e"]
remappedLetters = letters.map do |number|
	localCopyOfTheseLetters = letters.clone
	letters.shift
	localCopyOfTheseLetters
end

remappedLetters.each do |series|
	puts series.join " "
end

File: js/arrays/map/changeOriginal/js/closure.js
// closure.js

var letters = ["a","b","c","d","e"];
var remappedLetters = letters.map(function(number,index){
	var localCopyOfTheseLetters = letters.slice();
	letters.shift();
	return localCopyOfTheseLetters;
});
remappedLetters.forEach(function(series){
	console.log(series.join(" "));
});	